### PR TITLE
docs: add Storybook links to individual component pages

### LINKED ISSUE #4008: [docs] Include link to Storybook from component pages
### Code of conduct

- [X] I agree to follow this project's code of conduct.

### Description of issue

> Would it be possible to get a link/card to the storybook, as well? I think that would be great for our consumers.

_Originally posted by @najikahalsema in https://github.com/adobe/spectrum-web-components/issues/3957#issuecomment-1928090860_

- gather list of this data as json to pass into 11ty
- design a good location for another link
- render the link onto the component page

### PR DESCRIPTION
**Pull request by TarunAdobe**
## Description

Include link to Storybook on the component pages.

## Related issue(s)

- fixes #4008

## Motivation and context

The link to our storybook is sort of "hidden" in our documentation. You have to scroll down the components list and then you find the storybook. Linking it on top would encourage consumers to look for examples in storybook also.

## How has this been tested?

-   [ ] _Test case 1_
    1. Go here
    2. Do this
-   [ ] _Test case 2_
    1. Go here
    2. Do this

## Types of changes

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [x] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist

-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [x] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [ ] I have added tests to cover my changes.
-   [x] All new and existing tests passed.
-   [x] I have reviewed at the Accessibility Practices for this feature, see: [Aria Practices](https://www.w3.org/TR/wai-aria-practices/)

## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.

### CODE REVIEW

**Comment by Westbrook** [comment]
IMG_6336

FYI.

File: projects/documentation/scripts/gather-storybook-urls.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+const targetFile = path.resolve(targetDir, 'storybookURLs.js');
+
+export const gatherStorybookURLs = async (flatPackageNames) => {
+    const links = {};
+
+    // Get the storybook URLs for each package
+    flatPackageNames.forEach((packageName) => {
+        const storybookURL = `https://opensource.adobe.com/spectrum-web-components/storybook/index.html?path=/story/${packageName}/`;
+        links[packageName] = storybookURL;
+    });
```
Comment (Westbrook) [inline projects/documentation/scripts/gather-storybook-urls.js:35]: We'd want to confirm that this calculation processes to a story for every package, specifically those in the `Tools` section.

File: projects/documentation/src/components/styles.css [hunk 2]
```diff
@@ -922,53 +922,40 @@ sp-link:not(:defined) {
...
-.demo-config > div:nth-last-child(3):nth-of-type(1) ~ div {
-    max-width: calc(33% - 16px);
+.demo-config {
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    border-left: 1px solid rgb(63, 63, 63);
+    justify-content: center;
+    padding: 16px;
+    gap: 16px;
```
Comment (Westbrook) [inline projects/documentation/src/components/styles.css:954]:
```suggestion
```

**Review by Westbrook — COMMENTED** [review]
Can we use https://caniuse.com/css-container-queries to make sure that this:

image

Displays like:

https://chore-storybook-link-component--spectrum-web-components.netlify.app/components/action-bar/

Until you can fit all cards in a single row?

File: projects/documentation/src/components/styles.css [hunk 3]
```diff
@@ -922,53 +922,40 @@ sp-link:not(:defined) {
     text-decoration: underline;
 }
 
-.demo-placement.with-demo {
+.demo-container {
     display: grid;
-    grid-template-areas:
-        'demo'
-        'links';
+    grid-template-columns: 3fr 1fr;
```
Comment (Westbrook) [inline projects/documentation/src/components/styles.css:927]:
```suggestion
    grid-template-columns:  1fr auto;
```
This ensures the config column never needs to scroll horizontally.

**Comment by TarunAdobe** [comment]
> Can we use https://caniuse.com/css-container-queries to make sure that this:
> Until you can fit all cards in a single row?

Updated!

File: projects/documentation/src/components/styles.css [hunk 4]
```diff
@@ -998,45 +984,24 @@ sp-link:not(:defined) {
...
 }
 
 .usage-links :is(h2, h3) {
     all: unset;
 }
 
-.demo-placement sp-icon,
-.demo-placement sp-icon svg {
+.usage-links sp-card {
+    flex-basis: calc(25%-16px);
```
Comment (Westbrook) [inline projects/documentation/src/components/styles.css:999]:
```suggestion
    flex-basis: calc(25% - 16px);
    flex-grow: 1;
```

File: projects/documentation/src/components/styles.css [hunk 5]
```diff
@@ -1045,46 +1010,33 @@ p:has([alt='See it on NPM!']) {
...
-    }
-
-    .demo-placement.with-demo.with-controls {
-        grid-template-areas:
-            'demo links'
-            'config config';
-        grid-template-rows: 218px minmax(86px, auto);
+@container (max-width: 848px) {
+    .usage-links sp-card {
+        flex-basis: calc(50%-16px);
```
Comment (Westbrook) [inline projects/documentation/src/components/styles.css:1015]:
```suggestion
        flex-basis: calc(50% - 16px);
```

File: projects/documentation/src/components/styles.css (hunk 2 above)
Comment (Westbrook) [reply projects/documentation/src/components/styles.css]: I'm not seeing this...

**Review by Westbrook — APPROVED** [review]
There would likely be a benefit to a deeper, more nuanced review of the responsive delivery of pages with 1/2/3/4 cards at various screen sizes here, however, this is a great iterative step forward and a quality addition to the content/structure of the documentation site. Thanks for working through the tricky bits of the CSS on this!

:shipit:

**Pull request merged**